python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java

```diff
@@ -95,14 +95,31 @@ static void resolveTypeHierarchy(ClassDef classDef, @Nullable Symbol symbol, Pyt
       return;
     }
     for (Argument argument : argList.arguments()) {
-      Symbol argumentSymbol = getSymbolFromArgument(argument);
-      if (argumentSymbol == null) {
+      if (!argument.is(Kind.REGULAR_ARGUMENT)) {
         classSymbol.setHasSuperClassWithoutSymbol();
+        continue;
+      }
+      addParentClass(pythonFile, symbolsByName, classSymbol, (RegularArgument) argument);
+    }
+  }
+
+  private static void addParentClass(PythonFile pythonFile, Map<String, Symbol> symbolsByName, ClassSymbolImpl classSymbol, RegularArgument regularArgument) {
+    Name keyword = regularArgument.keywordArgument();
+    if (keyword != null) {
+      if (keyword.name().equals("metaclass")) {
+        classSymbol.setHasMetaClass();
       } else {
-        Symbol normalizedArgumentSymbol = normalizeSymbol(argumentSymbol, pythonFile, symbolsByName);
-        if (normalizedArgumentSymbol != null) {
-          classSymbol.addSuperClass(normalizedArgumentSymbol);
-        }
+        classSymbol.setHasSuperClassWithoutSymbol();
```

Comment:
it looks like keyword arguments don't alter type hierarchy (at least not explictely).
They are instead passed to `__init_subclass__()` method

https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__

Note that in python 2 they are a syntax error.